extensions/oidc-token-propagation-reactive/runtime/src/main/java/io/quarkus/oidc/token/propagation/reactive/OidcTokenPropagationReactiveConfig.java

```diff
@@ -0,0 +1,36 @@
+package io.quarkus.oidc.token.propagation.reactive;
+
+import java.util.Optional;
+
+import io.quarkus.runtime.annotations.ConfigItem;
+import io.quarkus.runtime.annotations.ConfigPhase;
+import io.quarkus.runtime.annotations.ConfigRoot;
+
+@ConfigRoot(name = "oidc-token-propagation-reactive", phase = ConfigPhase.BUILD_AND_RUN_TIME_FIXED)
+public class OidcTokenPropagationReactiveConfig {
+    /**
+     * Exchange the current token with OpenId Connect Provider for a new token using either
+     * "urn:ietf:params:oauth:grant-type:token-exchange" or "urn:ietf:params:oauth:grant-type:jwt-bearer" token grant
+     * before propagating it.
+     */
+    @ConfigItem(defaultValue = "false")
+    public boolean exchangeToken;
+
+    /**
+     * If the current token needs to be exchanged before being propagated then a
+     * `urn:ietf:params:oauth:grant-type:token-exchange` or `urn:ietf:params:oauth:grant-type:jwt-bearer` token grant
+     * specific property for passing the current token has to be used.
+     * By default it is set to `subject_token` which is required for `urn:ietf:params:oauth:grant-type:token-exchange` grant.
+     * Set it to `assertion` if you have to use "urn:ietf:params:oauth:grant-type:jwt-bearer" grant to exchange the token.
+     */
+    @ConfigItem(defaultValue = "subject_token")
+    public boolean exchangeTokenProperty;
```

Comment:
@gastaldi Sorry, I stepped outside I can probably avoid introducing this property and simply deduce its value from the configured grant name, and along the way also make sure a correct grant is configured if the token exchange is expected, I'll have a look now

Reply:
@gastaldi I was able to get rid of the property for both the reactive and class client token propagation filters